### DISCUSSION TITLE
Redis PubSub

### DISCUSSION THREAD

**Discussion by sgentile** [MichalLytek/type-graphql#1269] [Q&A] [answered]
I am using redis pubsub - I have a 'notification service' that is injected into my mutations.

I don't want to create my @PubSub at a resolver level, instead I want to have that just inside my notification service.   Is there a way to do that instead of using @PubSub 
ie.

this is what we have today
```
@Mutation(() => OrganizationLike)
    @Authorized()
    @Transactional()
    async likeOrganization(
        @PortalPubSub('notifications') publish: Publisher<NotificationSubscription>,
        @Ctx('user') user: User,
        @Arg('organization') organizationId: string
    ) {
        return this.organizationLikeService.likeOrganization(user, organizationId, publish);
    }
```

I'd rather just inject/use  my Organization service without passing this around.

**Comment by MichalLytek** [comment]
Create pubsub manually and put it into graphql context or dependency injection container

**Comment by sgentile** [comment]
In addition to adding to the createSchema ?

currently doing this...

```
            ...,
            AdministrationResolver,
        ],
        container: Container,
        validate: false,
        authChecker,
        pubSub,
    });
    ```

**Comment by sgentile** [comment]
(I create the pubsub outside of this and pass to the createSchema, along with my typeid container)

**Comment by MichalLytek** [comment, marked as answer]
TypeGraphQL can't inject anything into your services methods. You either pass that manually from resolvers or rely on the dependency injection container.

**Comment by sgentile** [comment]
ok thanks you answered my question - much appreciated

**Comment by sgentile** [comment]
for anyone else:
(I'm using typedi)
```
pubSub = new RedisPubSub({
                ...
            });
            
            Container.set('pubsub', pubSub);
            ```
            
            ```
            @Service()
export class NotificationService {
    constructor(@Inject('pubsub') private readonly pubsub: RedisPubSub) {}
    
    ...
    
    await this.pubsub.publish(triggerKey, notificationSubscription);
    ```
